### ISSUE TITLE
Specific GroupBy + Sum is evaluated locally

### ISSUE THREAD

**Issue by AlekseyMartynov** [dotnet/efcore#12433]
Version: 2.1.1 stable
Provider: `Microsoft.EntityFrameworkCore.SqlServer`

The following query is synthetic and doesn't look particulary useful, but I'd like to know why it's not translated into SQL. Does it indicate an issue with the current `GroupBy` support?

```c#
_nwind.Orders
    .Select(o => new { o.OrderId, o.Freight }) // removing this enables translation
    .GroupBy(o => o.Freight)
    .Select(g => g.Sum(i => i.Freight))
    .ToArray();
```

```c#
[Table("Orders")]
public partial class Order {
    [Key]
    [Column("OrderID")]
    public int OrderId { get; set; }

    [Column(TypeName = "money")]
    public decimal? Freight { get; set; }

    // ...
}
```

Expected SQL is as simple as:
```sql
SELECT SUM(Freight)
FROM Orders
GROUP BY Freight
```

**Comment by ajcvickers** [comment]
Notes for triage: tested this and it fails for me. Repro code, stack trace, and query model below.
```C#
[Table("Orders")]
public class Order
{
    [Key]
    [Column("OrderID")]
    public int OrderId { get; set; }

    [Column(TypeName = "money")]
    public decimal? Freight { get; set; }
}

public class BloggingContext : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        optionsBuilder
            .UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=Test;ConnectRetryCount=0");
    }

    public DbSet<Order> Orders { get; set; }
}

public class Program
{
    public static void Main()
    {
        using (var context = new BloggingContext())
        {
            context.Database.EnsureDeleted();
            context.Database.EnsureCreated();

            context.Add(new Order { Freight = 10 });
            context.Add(new Order { Freight = 10 });
            context.Add(new Order { Freight = 10 });

            context.SaveChanges();
        }

        using (var context = new BloggingContext())
        {
            var orders = context.Orders
                .Select(o => new { o.OrderId, o.Freight })
                .GroupBy(o => o.Freight)
                .Select(g => g.Sum(i => i.Freight))
                .ToArray();
        }
    }
}
```
```
dbug: Microsoft.EntityFrameworkCore.Query[10101]
      => Microsoft.EntityFrameworkCore.Query.RelationalQueryModelVisitor
      Compiling query model:
      'from IGrouping<Nullable<decimal>, <>f__AnonymousType0<int, Nullable<decimal>>> g in
          (from Order o in DbSet<Order>
          select new <>f__AnonymousType0<int, Nullable<decimal>>(
              [o].OrderId,
              [o].Freight
          )).GroupBy([o].Freight, new <>f__AnonymousType0<int, Nullable<decimal>>(
                  [o].OrderId,
                  [o].Freight
              ))
      select
          (from <>f__AnonymousType0<int, Nullable<decimal>> i in [g]
          select [i].Freight).Sum()'
dbug: Microsoft.EntityFrameworkCore.Query[10104]
      => Microsoft.EntityFrameworkCore.Query.RelationalQueryModelVisitor
      Optimized query model:
      'from IGrouping<Nullable<decimal>, <>f__AnonymousType0<int, Nullable<decimal>>> g in
          (from Order o in DbSet<Order>
          select new <>f__AnonymousType0<int, Nullable<decimal>>(
              [o].OrderId,
              [o].Freight
          )).GroupBy([o].Freight, new <>f__AnonymousType0<int, Nullable<decimal>>(
                  [o].OrderId,
                  [o].Freight
              ))
      select
          (from <>f__AnonymousType0<int, Nullable<decimal>> i in [g]
          select [i].Freight).Sum()'
warn: Microsoft.EntityFrameworkCore.Query[20500]
      => Microsoft.EntityFrameworkCore.Query.RelationalQueryModelVisitor
      The LINQ expression 'Sum()' could not be translated and will be evaluated locally.

Unhandled Exception: System.ArgumentException: Value does not fall within the expected range.
   at System.SharedTypeExtensions.GetSequenceType(Type type)
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.VisitMainFromClause(MainFromClause fromClause, QueryModel queryModel)
   at Remotion.Linq.QueryModelVisitorBase.VisitQueryModel(QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.VisitQueryModel(QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryModelVisitor.VisitQueryModel(QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.SqlTranslatingExpressionVisitor.VisitSubQuery(SubQueryExpression expression)
   at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.SqlTranslatingExpressionVisitor.Visit(Expression expression)
   at Microsoft.EntityFrameworkCore.Query.ExpressionVisitors.RelationalProjectionExpressionVisitor.Visit(Expression expression)
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.VisitSelectClause(SelectClause selectClause, QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryModelVisitor.VisitSelectClause(SelectClause selectClause, QueryModel queryModel)
   at Remotion.Linq.QueryModelVisitorBase.VisitQueryModel(QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.VisitQueryModel(QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryModelVisitor.VisitQueryModel(QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.CreateQueryExecutor[TResult](QueryModel queryModel)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQueryCore[TResult](Expression query, IQueryModelGenerator queryModelGenerator, IDatabase database, IDiagnosticsLogger`1 logger, Type contextType)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass13_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.Execute[TResult](Expression query)
   at Remotion.Linq.QueryableBase`1.GetEnumerator()
   at System.Collections.Generic.LargeArrayBuilder`1.AddRange(IEnumerable`1 items)
   at System.Collections.Generic.EnumerableHelpers.ToArray[T](IEnumerable`1 source)
   at Program.Main() in C:\Stuff\TwoOneCore\TwoOneCore\Program.cs:line 204
dbug
```

**Comment by smitpatel** [comment]
@ajcvickers - `Freight` is ignored property in EF model of northwind. Unmapped property with anonymous projection is expected to fail. Can you verify if it is happening with mapped properties?

**Comment by ajcvickers** [comment]
@smitpatel Repro code is above. It's not using any pre-existing Northwind database or configuration. It's just using the entity types above in a clean database.

**Comment by ajcvickers** [comment]
@smitpatel to investigate this for possible patch, since the model and query look simple.

**Comment by smitpatel** [comment]
Duplicate of #11999

**Comment by divega** [comment]
@AlekseyMartynov we believe the issue you reported is fixed in 2.2 (it should now evaluate on the server). We want to double-check that you are not hitting the issue described in #11999.
